src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java

```diff
@@ -67,26 +68,22 @@ public static KubernetesProvisioningLimits get() {
      * @param podTemplate the pod template used to schedule the agent
      * @param numExecutors the number of executors (pretty much always 1)
      */
-    @SuppressFBWarnings(value="JLM_JSR166_UTILCONCURRENT_MONITORENTER", justification = "Trust me here")
-    public boolean register(@Nonnull KubernetesCloud cloud, @Nonnull PodTemplate podTemplate, int numExecutors) {
-        AtomicInteger globalCount = getGlobalCount(cloud.name);
-        AtomicInteger podTemplateCount = getPodTemplateCount(podTemplate.getId());
-        synchronized (globalCount) {
-            synchronized (podTemplateCount) {
-                if (globalCount.get() + numExecutors <= cloud.getContainerCap()) {
-                    if (podTemplateCount.get() + numExecutors <= podTemplate.getInstanceCap()) {
-                        int g = globalCount.addAndGet(numExecutors);
-                        int p = podTemplateCount.addAndGet(numExecutors);
-                        LOGGER.log(Level.FINEST, () -> cloud.name + " global limit: " + g + "/" + cloud.getContainerCap());
-                        LOGGER.log(Level.FINEST, () -> podTemplate.getName() + " template limit: " + p + "/" + podTemplate.getInstanceCap());
-                        return true;
-                    } else {
-                        Metrics.metricRegistry().counter(MetricNames.REACHED_POD_CAP).inc();
-                    }
-                } else {
-                    Metrics.metricRegistry().counter(MetricNames.REACHED_GLOBAL_CAP).inc();
-                }
+    public synchronized boolean register(@Nonnull KubernetesCloud cloud, @Nonnull PodTemplate podTemplate, int numExecutors) {
+        int newGlobalCount = getGlobalCount(cloud.name) + numExecutors;
+        if (newGlobalCount <= cloud.getContainerCap()) {
+            int newPodTemplateCount = getPodTemplateCount(podTemplate.getId()) + numExecutors;
+            if (newPodTemplateCount <= podTemplate.getInstanceCap()) {
+                cloudCounts.put(cloud.name, newGlobalCount);
+                LOGGER.log(Level.FINEST, () -> cloud.name + " global limit: " + newGlobalCount + "/" + cloud.getContainerCap());
+
+                podTemplateCounts.put(podTemplate.getId(), newPodTemplateCount);
+                LOGGER.log(Level.FINEST, () -> podTemplate.getName() + " template limit: " + newPodTemplateCount + "/" + podTemplate.getInstanceCap());
+                return true;
+            } else {
+                Metrics.metricRegistry().counter(MetricNames.REACHED_POD_CAP).inc();
```

Comment:
I've just been hit by this bug, and it would've been great if there had been a "FINEST" log here (and in the other `else` clause two lines down), to be able to tell what's happening by looking at the logs. As-is, if capacity is reached, the metric gets increased, but nothing is shown in the logs.

Reply:
I added your suggestion to my latest patch

Reply:
(btw, feel free to chime in https://issues.jenkins.io/browse/JENKINS-66484, it may help getting attention on the issue!)